scipy/_lib/_array_api.py

```diff
@@ -736,41 +776,83 @@ def wrapper(*args, **kwargs):
     return decorator
 
 
-def make_skip_xp_backends(*funs, capabilities_table=None):
+def _make_xp_pytest_marks(*funcs, capabilities_table=None):
     capabilities_table = (xp_capabilities_table if capabilities_table is None
                           else capabilities_table)
+    import pytest
+    from scipy._lib.array_api_extra.testing import lazy_xp_function
+
+    marks = []
+    for func in funcs:
+        capabilities = capabilities_table[func]
+        exceptions = capabilities['exceptions']
+        reason = capabilities['reason']
 
+        if capabilities['cpu_only']:
+            marks.append(pytest.mark.skip_xp_backends(
+                cpu_only=True, exceptions=exceptions, reason=reason))
+        if capabilities['np_only']:
+            marks.append(pytest.mark.skip_xp_backends(
+                cpu_only=True, exceptions=exceptions, reason=reason))
+
+        for mod_name, reason in capabilities['skip_backends']:
+            marks.append(pytest.mark.skip_xp_backends(mod_name, reason=reason))
+        for mod_name, reason in capabilities['xfail_backends']:
+            marks.append(pytest.mark.xfail_xp_backends(mod_name, reason=reason))
+
+        lazy_kwargs = {k: capabilities[k]
+                       for k in ('allow_dask_compute', 'jax_jit',
+                                 'static_argnums', 'static_argnames')}
+        lazy_xp_function(func, **lazy_kwargs)
+
+    return marks
+
+
+def make_skip_xp_backends(*funcs, capabilities_table=None):
```

Comment:
This is no longer just about skipping tests, as it also applies `lazy_xp_function`.
I will mass rename it to `make_xp_test_case` in a follow-up PR. I didn't to it here in the interest of keeping the PR readable.